miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/InstrumentStatusRestController.java

```diff
@@ -43,15 +41,16 @@ public List<InstrumentStatusDto> listAll() throws IOException {
           if (aRunning != bRunning) {
             return bRunning.ordinal() - aRunning.ordinal();
           }
-          LocalDate aDate = getCompareDate(a);
-          LocalDate bDate = getCompareDate(b);
-          if (bDate == null) {
-            return -1;
-          } else if (aDate == null) {
+          long idA = a.getId();
+          long idB = b.getId();
+          if (idA < idB) {
             return 1;
+          } else if (idA == idB) {
+            return 0;
+          } else {
+            return -1;
           }
-          return bDate.compareTo(aDate);
```

Comment:
* ID should be another level of sorting _after_ sorting by date - it should not replace sorting by date
* Can use `Long.compare(long, long)`